src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -342,156 +332,173 @@ async Task<bool> NextResult(bool async, bool isConsuming=false)
             IBackendMessage msg;
             Debug.Assert(!_isSchemaOnly);
 
-            // If we're in the middle of a resultset, consume it
-            switch (State)
+            try
             {
-            case ReaderState.BeforeResult:
-            case ReaderState.InResult:
-                await ConsumeRow(async);
-                while (true)
+                // If we're in the middle of a resultset, consume it
+                switch (State)
                 {
-                    var completedMsg = await Connector.ReadMessage(async, DataRowLoadingMode.Skip);
-                    switch (completedMsg.Code)
+                case ReaderState.BeforeResult:
+                case ReaderState.InResult:
+                    await ConsumeRow(async);
+                    while (true)
                     {
-                    case BackendMessageCode.CompletedResponse:
-                    case BackendMessageCode.EmptyQueryResponse:
-                        ProcessMessage(completedMsg);
+                        var completedMsg = await Connector.ReadMessage(async, DataRowLoadingMode.Skip);
+                        switch (completedMsg.Code)
+                        {
+                        case BackendMessageCode.CompletedResponse:
+                        case BackendMessageCode.EmptyQueryResponse:
+                            ProcessMessage(completedMsg);
+                            break;
+                        default:
+                            continue;
+                        }
```

Comment:
This could be rewritten using `if`:

```csharp
if (completedMsg.Code = BackendMessageCode.CompletedResponse ||
    completedMsg.Code = BackendMessageCode.EmptyQueryResponse)
{
    ProcessMessage(completedMsg);
    break;
}
else
    continue;
```

Not so important because saves only one line.

Reply:
The switch could actually be more efficient (possibly translated to a jump table by the compiler) :) Will leave as-is as you don't seem to mind.